packages/fiori/src/UploadCollectionItem.js

```diff
@@ -0,0 +1,380 @@
+ *
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import @ui5/webcomponents-fiori/dist/UploadCollectionItem.js";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.fiori.UploadCollectionItem
+ * @extends UI5Element
+ * @tagname ui5-upload-collection-item
+ * @public
+ */
+class UploadCollectionItem extends ListItem {
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return [ListItem.styles, UploadCollectionItemCss];
+	}
+
+	static get template() {
+		return UploadCollectionItemTemplate;
+	}
+
+	static async onDefine() {
+		await Promise.all([
+			Button.define(),
+			Icon.define(),
+			Input.define(),
+			Link.define(),
+			Label.define(),
+			fetchI18nBundle("@ui5/webcomponents-fiori"),
+		]);
+	}
+
+	constructor() {
+		super();
+		this.i18nBundle = getI18nBundle("@ui5/webcomponents-fiori");
+	}
+
+	onBeforeRendering() {
+		if (!this.focused) {
+			this._editing = false;
+		}
+	}
+
+	onAfterRendering() {
+		if (this.focused && this._editing) {
+			this.focusAndSelectText();
+		}
+	}
+
+	async focusAndSelectText() {
+		await this.focus();
+
+		const inp = this.shadowRoot.getElementById("ui5-uci-edit-input");
+
+		if (inp.getFocusDomRef()) {
+			inp.getFocusDomRef().setSelectionRange(0, this._fileNameWithoutExtension.length);
+		}
+	}
+
+	/**
+	 * @override
+	 */
+	onDetailClick(event) {
+		super.onDetailClick(event);
+		this._editing = true;
+	}
+
+	_onInputChange(event) {
+		if (this.shadowRoot.getElementById("ui5-uci-edit-cancel").active) {
+			return;
+		}
+
+		this._editing = false;
+		this.fileName = event.target.value + this._fileExtension;
+		this.fireEvent("rename");
+	}
+
+	_onRenameCancel(event) {
+		this._editing = false;
+	}
+
+	_onFileNameClick(event) {
+		this.fireEvent("fileNameClick");
+	}
+
+	_onRetry(event) {
+		this.fireEvent("retry");
+	}
+
+	_onTerminate(event) {
+		this.fireEvent("terminate");
+	}
+
+	/**
+	 * @override
+	 */
+	get classes() {
+		const result = super.classes;
+
+		return {
+			main: {
+				...result.main,
+				"ui5-uci-root": true,
+				"ui5-uci-root-editing": this._editing,
+				"ui5-uci-root-uploading": this.uploadState === UploadState.Uploading,
+			},
+			progressIndicator: {
+				"ui5-uci-progress-indicator": true,
+				"error": this.uploadState === UploadState.Error,
+			},
+		};
+	}
+
+	get styles() {
+		return {
+			progressBar: {
+				"flex-basis": `${this.progress}%`,
+			},
+			progressBarRemaining: {
+				"border-left": this.progress !== 0 ? "none" : "",
+				"border-radius": this.progress === 0 ? "0.5rem" : "0 0.5rem 0.5rem 0",
+			},
+		};
+	}
+
+	/**
+	 * @override
+	 */
+	get modeDelete() {
+		return !this.noDelete && super.modeDelete;
+	}
+
+	get _fileNameWithoutExtension() {
+		return this.fileName.substring(0, this.fileName.length - this._fileExtension.length);
+	}
+
+	get _fileExtension() {
+		return this.fileName.includes(".") ? `.${this.fileName.split(".").pop()}` : "";
```

Comment:
this will have a problem with files such as `.gitignore` that start with a dot and have no extension
Please research how to get the extension of a file safely and once you find the algorithm, implement it in: `base/src/utils/getFileExtension.js`
and let it export a single method
`export default getFileExtension;`
We might need it elsewhere in the future too :)

Reply:
Done